test/test_dumpers.py

```diff
@@ -0,0 +1,86 @@
+"""Test dumping works correcly."""
+
+import io
+import json
+from functools import partial
+from pathlib import Path
+
+import pytest
+
+from castep_outputs.cli.castep_outputs_main import parse_all, parse_single
+from castep_outputs.utilities.utility import normalise
+
+_DATA_FOLDER = Path(__file__).parent / "data_files"
+
+def _to_complex(complx):
+    if (isinstance(complx, dict) and not complx.keys() ^ {"real", "imag"}):
+        return complex(**complx)
+    return complx
+
+@pytest.mark.parametrize("file,typ,ref", [
+    ("test.cell", "cell", "cell"),
+    ("test.param", "param", "param"),
+    ("test.castep", "castep", "castep"),
+    ("pp-md.castep", "castep", "pp-md"),
+    ("si8-md.castep", "castep", "si8-md"),
+    ("test.bands", "bands", "bands"),
+    ("test.md", "md", "md"),
+    ("test.elastic", "elastic", "elastic"),
+    ("test.ts", "ts", "ts"),
+    ("test.chdiff_fmt", "chdiff_fmt", "chdiff_fmt"),
+    ("test.den_fmt", "den_fmt", "den_fmt"),
+    ("test.pot_fmt", "pot_fmt", "pot_fmt"),
+    ("test.elf_fmt", "elf_fmt", "elf_fmt"),
+    ("test.xrd_sf", "xrd_sf", "xrd_sf"),
+    ("test.phonon_dos", "phonon_dos", "phonon_dos"),
+    ("test.efield", "efield", "efield"),
+    ("test.magres", "magres", "magres"),
+    ("test.tddft", "tddft", "tddft"),
+    ("test.err", "err", "err"),
+    ("test.phonon", "phonon", "phonon"),
+])
+@pytest.mark.parametrize("out_format", ("json", "ruamel", "pyyaml"))
+def test_dump(file, typ, ref, out_format):
+    """Test dumpers work correctly."""
+    # Get loaders
+    if out_format == "json":
+        loader = json.load
+    elif out_format == "ruamel":
+        module = pytest.importorskip("ruamel.yaml")
+        yaml_eng = module.YAML(typ="unsafe")
+        loader = yaml_eng.load
+    elif out_format == "pyyaml":
+        module = pytest.importorskip("yaml")
+        try:
+            from yaml import CLoader as Loader
+        except ImportError:
+            from yaml import Loader
```

Comment:
Version 6 of pyyaml has wheels on PyPI; these presumably provide CLoader. Do we really need this fallback for the test suite?

Reply:
I doubt it, but it's very little visual noise for the safety. Guess I could do:
```suggestion
        module = pytest.importorskip("yaml")
        Loader = module.CLoader if hasattr(module, "CLoader") else module.Loader
```

Reply:
If it's important to have the tests work with an old or source version of pyyaml, then leave as-is.

Otherwise you might as well
`Loader = module.CLoader` and let the `NameError` happen.

Reply:
Does seem like `Loader` and `CLoader` have (slightly) different behaviour. And that some builds don't include CLoader (as of 2020?)

https://stackoverflow.com/questions/47715566/cannot-load-cloader-with-pyyaml/62543781#62543781

Reply:
Happy to leave as-is then